Src/BeebMem.cpp

```diff
@@ -286,7 +288,12 @@ unsigned char BeebReadMem(int Address) {
 	unsigned char Value = 0xff;
 
 	if (MachineType == Model::B) {
-		if (Address >= 0x8000 && Address < 0xc000) return Roms[ROMSEL][Address - 0x8000];
+		if (Address >= 0x8000 && Address < 0xc000) {
+			if (ROMSEL > 3)
+				return Roms[ROMSEL][Address - 0x8000];
+  			if (ROMSEL < 4)
```

Comment:
Is this a hardware behaviour that is being emulated? Would an extended ROM not work in bank 12, for example. https://stardot.org.uk/forums/viewtopic.php?p=291695#p291695

Reply:
Sorry, I should have added more explanation.  Technically there is nothing limiting all of the roms being extended. I decided to limit it the functionality to roms 0-3 for all machines because I felt that it was likely that no more than 4 would be used at anyone time and didn’t want to reserve space I felt would not get used.  Maybe I should have used a named constant instead of the literal values.

Reply:
I have actually tested this now, and for rom slot >7 it actually gives an access violation error message when accessing the IDEdrive stuff.  So I would suggest that it be limited to 7 potential slots.

Reply:
That sounds a bit strange. I'm thinking there may be away to enable extended roms in any slot, but would be a bigger code change.